### ISSUE TITLE
Shields & Rewards Bar Buttons Are Not VoiceOver Accessible and Popover Not Dismissible

### ISSUE THREAD

**Issue by JonKaleidoscope** [brave/brave-ios#2365]
### Description: 
The Brave Shields and Rewards buttons on the tool bar are not accessible when using VoiceOver. When either of these menus are open, the popover views trap VoiceOver users on the screen because there is no way to dismiss/close the view when VoiceOver is running(the swipe away gesture not VoiceOver friendly). 

### Steps to Reproduce 
Buttons Not Accessible 
  1. Enable VoiceOver and Launch Brave
  2. Swipe to navigate to the "Shields" button

Popover trapping users on screen:
  1. Enable Accessibility Shortcuts (shortcut for VoiceOver)
  2. Launch Brave with VoiceOver disabled
  3. Tap on "Shields" Button
  4. Enable VoiceOver with shortcut and try to close Shields screen

**Actual result:** 
- Button not accessible
- Unable to dismiss view

**Expected result:**
- Buttons are accessible when using VoiceOver
- VoiceOver users are able to close view

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easily

**Brave Version:** 

**Device details:** 
iPhone *

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information
All uses of the PopoverController have similar issues with use when opening trapping users on the current screen.
Suggested Fixes:
  1. Add Close `X` on Popovers. By Adding the button the views will inherit VoiceOver controls for free from the button.
  2. Unable the background View with a VoiceOver action to close the view. All Popover views would inherit capabilities to dismiss itself for free.

Initial work start in the following PR: https://github.com/brave/brave-ios/pull/2315
